python/pyspark/sql/datasource.py

```diff
@@ -426,6 +426,10 @@ def read(self, partition: InputPartition) -> Iterator[Union[Tuple, Row]]:
             in the final DataFrame.
         """
         ...
+        raise PySparkNotImplementedError(
```

Comment:
@chaoqin-li1123 
nit: if not needed, let's remove @abstract in the method definition.
Please do this as follow-up PR. I'd like to unblock this PR to let you move forward.